[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta

wz337 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114710

Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @kiukchung @d4l3k @lucasllc

wz337 committed:
[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta

Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

[ghstack-poisoned]

wanchaol commented on docs/source/distributed.device_mesh.rst:
```diff
@@ -0,0 +1,5 @@
+DeviceMesh
+========================
+
+.. automodule:: torch.distributed.device_mesh
```
hmmm, I think this would add a new entry to the left bar of the documentation?

I am thinking of whether we should add this as a separate section under `distributed.rst` or we should add it as a separate page

wz337 replied:
@wconstab 

Based on our discussion, we prefer to add `init_device_mesh` and `DeviceMesh` under the current distributed doc page (https://pytorch.org/docs/stable/distributed.html) instead of having a separate .rst page. Want to ask your opinion about it.

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wanchaol commented on docs/source/distributed.rst:
```diff
@@ -339,6 +341,18 @@ an opaque group handle that can be given as a ``group`` argument to all collecti
+distributed process group easily. :func:`~torch.distributed.device_mesh` function can be used to
+create new DeviceMesh, with a mesh tensor describing the device topology.
+
+.. autofunction:: DeviceMesh
```
I think this should be autoclass instead of autofunction?

wanchaol commented on docs/source/distributed.rst:
```diff
@@ -339,6 +341,18 @@ an opaque group handle that can be given as a ``group`` argument to all collecti
+DeviceMesh is a higher level abstraction that manages process groups (or NCCL communicators).
+It allows user to easily create inter node and intra node process groups without worrying about
+how to set up the ranks correctly for different sub process groups, and it helps manage those
+distributed process group easily. :func:`~torch.distributed.device_mesh` function can be used to
```
I think this should be `~torch.distributed.init_device_mesh`?

wanchaol reviewed (approved):
lgtm, some nit comments for the doc, thanks for working on this!

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename _device_mesh.py to device_mesh.py to prepare for beta"


Rename _device_mesh.py to device_mesh.py, update all callsites, adds documentation.

Differential Revision: [D51629761](https://our.internmc.facebook.com/intern/diff/D51629761/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]
